Detectors/ITSMFT/MFT/reconstruction/src/TrackerTask.cxx

```diff
@@ -0,0 +1,99 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file  TrackerTask.cxx
+/// \brief Task driving the track finding from MFT clusters
+/// \author bogdan.vulpescu@cern.ch
+/// \date Feb. 7, 2018
+
+#include "MFTReconstruction/TrackerTask.h"
+#include "DataFormatsITSMFT/Cluster.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+#include "SimulationDataFormat/MCTruthContainer.h"
+
+ClassImp(o2::MFT::TrackerTask)
+
+  using namespace o2::MFT;
+using namespace o2::Base;
+
+//_____________________________________________________________________________
+TrackerTask::TrackerTask(Int_t n, Bool_t useMCTruth) : FairTask("MFTTrackerTask"), mTracker(n)
+{
+
+  if (useMCTruth) {
+    mTrkLabels = new o2::dataformats::MCTruthContainer<o2::MCCompLabel>;
+  }
+}
+
+//_____________________________________________________________________________
+TrackerTask::~TrackerTask()
+{
+
+  if (mTracksArray) {
+    mTracksArray->clear();
+    delete mTracksArray;
+  }
+  if (mTrkLabels) {
+    mTrkLabels->clear();
+    delete mTrkLabels;
+  }
+}
+
+//_____________________________________________________________________________
+InitStatus TrackerTask::Init()
+{
+
+  FairRootManager* mgr = FairRootManager::Instance();
+  if (!mgr) {
+    LOG(ERROR) << "Could not instantiate FairRootManager. Exiting ..." << FairLogger::endl;
+    return kERROR;
+  }
+
+  mClustersArray = mgr->InitObjectAs<const std::vector<o2::ITSMFT::Cluster>*>("MFTCluster");
+  if (!mClustersArray) {
+    LOG(ERROR) << "MFT clusters not registered in the FairRootManager. Exiting ..." << FairLogger::endl;
+    return kERROR;
+  }
+
+  // Register output container
+  mgr->RegisterAny("MFTTrack", mTracksArray, kTRUE);
```

Comment:
maybe you should also check that the `mTrackArray` pointer is valid. Should be valid after this operation, @sawenzel ? Not sure if I have understood the `TypeAddressPair` and register magic correct.